### PR TITLE
Fix sync requests on startup

### PR DESCRIPTION
**Pull request by sasurobert**

### CODE REVIEW

File: dataRetriever/requestHandlers/requestHandler.go [hunk 1]
```diff
@@ -242,7 +242,7 @@ func (rrh *resolverRequestHandler) RequestShardHeader(shardID uint32, hash []byt
 		return
 	}
 
-	log.Trace("requesting shard header from network",
+	log.Debug("requesting shard header from network",
```
Comment (raduchis) [inline dataRetriever/requestHandlers/requestHandler.go]: Is the Debug needed?

File: node/node.go [hunk 2]
```diff
@@ -252,6 +252,12 @@ func (n *Node) StartConsensus(epoch uint32) error {
 	}
 
 	bootstrapper.StartSync()
+	epoch := uint32(0)
+	crtBlockHeader := n.blkc.GetCurrentBlockHeader()
+	if !check.IfNil(crtBlockHeader) {
+		epoch = crtBlockHeader.GetEpoch()
+		log.Debug("starting consensus", "epoch", epoch)
```
Comment (raduchis) [inline node/node.go]: I would put log.Info here, because it happens just once when you start the node

File: process/track/blockProcessor.go [hunk 3]
```diff
@@ -311,14 +311,19 @@ func (bp *blockProcessor) getHighestRoundInReceivedHeaders(
 }
 
 func (bp *blockProcessor) requestHeadersIfNothingNewIsReceived(
+	lastNotarizedHeader data.HeaderHandler,
```
Comment (raduchis) [inline process/track/blockProcessor.go]: lastNotarized header is not needed here, just the lastNotarizedHeader.GetNonce(), just the nonce could be passed as param.

File: dataRetriever/requestHandlers/requestHandler.go (hunk 1 above)
Comment (iulianpascalau) [reply dataRetriever/requestHandlers/requestHandler.go]: yes, for the moment, we want to see the effective requests

File: node/node.go (hunk 2 above)
Comment (iulianpascalau) [reply node/node.go]: done

File: process/track/blockProcessor.go (hunk 3 above)
Comment (iulianpascalau) [reply process/track/blockProcessor.go]: done

File: node/node.go [hunk 4]
```diff
@@ -252,6 +252,12 @@ func (n *Node) StartConsensus(epoch uint32) error {
 	}
 
 	bootstrapper.StartSync()
+	epoch := uint32(0)
+	crtBlockHeader := n.blkc.GetCurrentBlockHeader()
+	if !check.IfNil(crtBlockHeader) {
+		epoch = crtBlockHeader.GetEpoch()
+		log.Info("starting consensus", "epoch", epoch)
```
Comment (bogdan-rosianu) [inline node/node.go]: shouldn't this log message be displayed in both cases ?

File: node/node_test.go [hunk 5]
```diff
@@ -2406,9 +2406,9 @@ func TestNode_SendBulkTransactionsMultiShardTxsShouldBeMappedCorrectly(t *testin
 			}
 			for _, txBuff := range txsBuff {
 				tx := transaction.Transaction{}
-				err := marshalizer.Unmarshal(&tx, txBuff)
-				if err != nil {
-					assert.Fail(t, err.Error())
+				errMarshal := marshalizer.Unmarshal(&tx, txBuff)
+				if errMarshal != nil {
```
Comment (bogdan-rosianu) [inline node/node_test.go]: `assert.Nil(t, errMarshal)` or even `require`

File: node/node.go (hunk 4 above)
Comment (iulianpascalau) [reply node/node.go]: :1st_place_medal: done

File: node/node_test.go (hunk 5 above)
Comment (iulianpascalau) [reply node/node_test.go]: require

File: dataRetriever/requestHandlers/requestHandler.go (hunk 1 above)
Comment (raduchis) [reply dataRetriever/requestHandlers/requestHandler.go]: ok

File: node/node.go (hunk 2 above)
Comment (raduchis) [reply node/node.go]: ok

File: process/track/blockProcessor.go (hunk 3 above)
Comment (raduchis) [reply process/track/blockProcessor.go]: ok

**Pull request merged**